docs/content/latest/explore/ysql-language-features/gin.md

```diff
@@ -0,0 +1,501 @@
+
+## Compatible types
+
+GIN indexes can only be created over a few types:
+
+- a GIN index on a tsvector column indexes text elements
+- a GIN index on a array column indexes array elements
+- a GIN index on a jsonb column indexes keys/values
+
+With extensions, more types can be supported.
+However, extension support is still in progress:
+
+- [ ] btree_gin
+- [ ] hstore
+- [x] pg_trgm
+
+## Grammar
+
+### CREATE INDEX
+
+Create the index using `USING ybgin` to specify the index access method.
+
+`CREATE INDEX ON mytable USING ybgin (mycol);`
+
+The `gin` access method is reserved for temporary relations while `ybgin` is for Yugabyte-backed relations.
+You may still specify `USING gin`, and, if `mytable` is not a temporary table, it will be automatically substituted for `ybgin`.
+
+Limitations:
+
+- Multicolumn GIN indexes are not supported: this can be done in the future.
+- Unique indexes are not supported: this doesn't make sense.
+- ASC/DESC/HASH cannot be specified: it defaults to ASC for prefix match purposes, so this can be relaxed in the future.
+
+### DELETE, INSERT, UPDATE
+
+Writes are fully supported.
+
+Note that UPDATEs may be expensive since they are currently implemented as DELETE + INSERT.
+
+### SELECT
+
+Only [certain SELECTs][operators] use the GIN index.
+Even among them, there are currently some limitations:
+
+- Special scans are not supported.
+  - Scans cannot include rows whose column has zero index keys.
+  - Scans cannot scan the entire index.
+- Scans cannot ask for more than one index key.
+  For example, a request for all rows whose array contains elements 1 or 3 will fail, but one that asks for elements 1 _and_ 3 can succeed by choosing one of the elements for index scan and rechecking the entire condition later.
+  However, the choice between 1 and 3 is currently unoptimized, so 3 may be chosen even though 1 corresponds to less rows.
+- Recheck is always done rather than on a case-by-case basis, meaning there can be an unnecessary performance penalty.
+
+If a query is unsupported, you may avoid the error by disabling index scan: `SET enable_indexscan TO off;` before the query and `SET enable_indexscan TO on;` afterwards.
+In the near future, cost estimates should route such queries to sequential scan.
+
+[operators]: https://www.postgresql.org/docs/13/gin-builtin-opclasses.html
+
+## Changes from upstream PostgreSQL
+
+For those familiar with upstream PostgreSQL GIN indexes, Yugabyte GIN indexes are slightly different.
+
+- Upstream PostgreSQL uses bitmap index scan; Yugabyte uses index scan.
+- Deletes to the index are written explicitly: this is due to storage-layer architecture differences, and it's also true for regular indexes.
+- Fast update is not supported: this isn't practical for a distributed, log-structured database.
+- Fuzzy search limit is not supported: it can be in the future.
+
+## Examples
+
+### Setup
+
+To begin, I set up a cluster using `yb-ctl`.
```

Comment:
nit: can skip the `I`. If you want to emphasize the user choice you can maybe say:
```
To begin set up a YugabyteDB cluster.
For instance, using `yb-ctl` .. 
```